### DISCUSSION TITLE
IconTintColorBehavior

### DISCUSSION THREAD

**Discussion by Avots** [CommunityToolkit/Maui#512] [Q&A]
Hi!
I have a list with icons of various colors.
How do I make the images have their color, passed by binding?
I tried this way but it didn't work
    <Image.Behaviors>
           <mct:IconTintColorBehavior TintColor="{Binding ImageColor, Converter={StaticResource StringToColor}}" />
   </Image.Behaviors>

**Comment by pictos** [comment]
@Avots can you share the xaml of the list as well?

**Reply by Avots** [reply]
[MVVMDemoWithMAUI.zip](https://github.com/CommunityToolkit/Maui/files/9170108/MVVMDemoWithMAUI.zip)

In this example I added the ones that works.
With Binding doesn't work!
Can you help me?
![image](https://user-images.githubusercontent.com/79453107/180508406-b04248f0-9cce-49a1-8e87-3bc9137d2416.png)

**Reply by pictos** [reply]
@Avots sorry for the late reply, but AFAIK `Behavior`s don't inherit the BindingContext for the control that implements it, so you need to set the BindingContext by yourself or use the _long_ binding syntax, you can see a sample [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/samples/XCT.Sample/Pages/Behaviors/CharactersValidationBehaviorPage.xaml#L57)

**Reply by Avots** [reply]
Thanks @pictos!

Will there be a way to use equal to xamarin.forms?

xmlns:xct="http://xamarin.com/schemas/2020/toolkit"

<Image Source="image.png"
        xct:IconTintColorEffect.TintColor="{Binding ImageColor}"/>

**Reply by pictos** [reply]
@Avots, yes

`xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"`

**Reply by Avots** [reply]
@pictos Sorry, missed the other part.
I meant to use it directly in the image equal in xamarin!

<Image Source="troubleshoot.png"
      xct:IconTintColorEffect.TintColor="{Binding ImageColor}"/>

In xamarin it works!
![image](https://user-images.githubusercontent.com/79453107/182833417-33d11bc5-6441-43a9-a7fc-105f991e3910.png)

Thank you again!
I will see what I can do!

**Reply by pictos** [reply]
Ooh, so we can't, due to some changes on architecture we don't use Attached properties for this feature anymore

**Comment by CliffAgius** [comment]
Closed as answered
